From 58b8efbec706cfe9d57ed0404eb71beedc7d1af8 Mon Sep 17 00:00:00 2001
From: oreo639 <oreo6391@gmail.com>
Date: Tue, 29 Jul 2025 23:41:37 -0700
Subject: [PATCH] remove asm.S and neon.S from _SOURCES

Previously asm.S and neon.S were getting assembled into object files and linked.
While for asm.S, this doesn't cause any problems, neon.S never ends up including
a .note.GNU-stack section resulting in the final library being marked as
requiring executable stack.

These files just define macros used by cpudetect.S and hevcdsp_qpel_neon.S so
them getting assembled and linked individually doesn't seem intended.
---
 libde265/arm/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libde265/arm/Makefile.am b/libde265/arm/Makefile.am
index 9ef62d98d..819ccf20e 100644
--- a/libde265/arm/Makefile.am
+++ b/libde265/arm/Makefile.am
@@ -26,10 +26,10 @@ if ENABLE_ARM_THUMB
 endif
 
 libde265_arm_neon_la_SOURCES = \
-	asm.S \
 	cpudetect.S \
-	hevcdsp_qpel_neon.S \
-	neon.S
+	hevcdsp_qpel_neon.S
+
+EXTRA_DIST = asm.S neon.S
 
 if HAVE_VISIBILITY
 	libde265_arm_neon_la_CXXFLAGS += -DHAVE_VISIBILITY
